### PR TITLE
[Monitor] add az monitor data-collection commands

### PR DESCRIPTION
**Pull request by kairu-ms**
Close https://github.com/Azure/azure-cli/issues/15970
Close https://github.com/Azure/azure-cli/issues/15543

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Monitor

File: src/monitor-control-service/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+Install this extension using the below CLI command
+```
+az extension add --name data-collection
+```
+
+### Included Features ###
+#### data-collection data-collection-rule-association ####
+##### Create #####
+```
+az data-collection data-collection-rule-association create --association-name "myAssociation" \
```
Comment (haroldrandom) [inline src/monitor-control-service/README.md:14]: This line is wrong and below

File: src/monitor-control-service/azext_amcs/manual/action.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        super(AddDataFlows, self).__call__(parser, namespace, action, option_string)
+
+    def get_action(self, values, option_string):  # pylint: disable=no-self-use
+        try:
+            properties = defaultdict(list)
+            for (k, v) in (x.split('=', 1) for x in values):
+                properties[k].append(v)
+            properties = dict(properties)
+        except ValueError:
+            raise CLIError('usage error: {} [KEY=VALUE ...]'.format(option_string))
```
Comment (haroldrandom) [inline src/monitor-control-service/azext_amcs/manual/action.py:25]: should raise errors from azure.cli.core.azclierror

**Review by haroldrandom — CHANGES_REQUESTED** [review]
please fix comments above

File: src/monitor-control-service/_az_debug/clicommon-flatten-object-map.txt [hunk 3]
```diff
@@ -0,0 +1,28 @@
+data_collection_rule_association_proxy_only_resource_list_result<2>
+data_collection_rule_association_proxy_only_resource<7>
```
Comment (jsntcy) [inline src/monitor-control-service/_az_debug/clicommon-flatten-object-map.txt:2]: what's `_az_debug`?

Comment (kairu-ms) [reply src/monitor-control-service/_az_debug/clicommon-flatten-object-map.txt]: removed. Thanks~

**Review by haroldrandom — APPROVED** [review]

**Pull request merged**